### ISSUE TITLE
clone fails to enable ORA special remote with git-annex 7.x

### ISSUE THREAD

**Issue by mih** [datalad/datalad#4509]
Found with 0.13rc1 installed via conda on an HPC system

```
$  datalad clone ria+file:///p/largedata/.../enki#~bids_raw 
[INFO   ] Scanning for unlocked files (this may take some time)                          
[INFO   ] Cannot determine special remote name, got: '' 
install(ok): /.../bids_raw (dataset)
$ cd bids_raw/
$ git annex info
...
        9de58e19-34aa-4bbc-8c59-a8b2549b0b0e -- origin
untrusted repositories: 1
        c1806013-fe09-4770-b04c-e71f29acf899 -- judac-storage
...
$ git cat-file -p git-annex:remote.log
c1806013-fe09-4770-b04c-e71f29acf899 archive-id=f380eccc-643b-11ea-a42a-ac1f6bac9bd2 autoenable=true encryption=none externaltype=ora name=judac-storage type=external url=ria+ssh://judac.fz-juelich.de/p/.../enki timestamp=1588933033.985768224s
```

The config in the store says:

```
[core]
        repositoryformatversion = 0
        filemode = true
        bare = true
[datalad "ora-remote"]
        uuid = c1806013-fe09-4770-b04c-e71f29acf899
[annex]
        uuid = 9de58e19-34aa-4bbc-8c59-a8b2549b0b0e
        version = 7
```

This looks like a genuine bug @bpoldrack

**Comment by bpoldrack** [comment]
Agree. That's clearly a bug.
Note to myself: Look at this:
```
[INFO   ] Cannot determine special remote name, got: '' 
```

**Comment by bpoldrack** [comment]
Declaring "important", since to not have this outcome is the entire point of recent change to `clone` and why there's 
```
[datalad "ora-remote"]
        uuid = c1806013-fe09-4770-b04c-e71f29acf899
```
added by `create-sibling-ria` in the first place.

**Comment by mih** [comment]
This is also hard to fix. Even a manual URL override doesn't help.

```
$ git annex enableremote judac-storage
enableremote judac-storage 
git-annex: Cannot determine special remote name, got: ''
failed
git-annex: enableremote: 1 failed

$ git annex enableremote judac-storage url=ria+file:///p/largedata/.../enki
enableremote judac-storage 
git-annex: Cannot determine special remote name, got: ''
failed
git-annex: enableremote: 1 failed
```

**Comment by mih** [comment]
```
$ git annex --debug enableremote judac-storage url=ria+file:///p/largedata/.../enki
[2020-05-10 14:01:02.855444] read: git ["--git-dir=.git","--work-tree=.","--literal-pathspecs","show-ref","git-annex"]
[2020-05-10 14:01:02.862236] process done ExitSuccess
[2020-05-10 14:01:02.862468] read: git ["--git-dir=.git","--work-tree=.","--literal-pathspecs","show-ref","--hash","refs/heads/git-annex"]
[2020-05-10 14:01:02.868764] process done ExitSuccess
[2020-05-10 14:01:02.869479] read: git ["--git-dir=.git","--work-tree=.","--literal-pathspecs","log","refs/heads/git-annex..d962944f1a33d127a1feade76e48cdad67b159f0","--pretty=%H","-n1"]
[2020-05-10 14:01:02.876412] process done ExitSuccess
[2020-05-10 14:01:02.877227] chat: git ["--git-dir=.git","--work-tree=.","--literal-pathspecs","cat-file","--batch"]
[2020-05-10 14:01:02.897196] chat: git ["--git-dir=.git","--work-tree=.","--literal-pathspecs","cat-file","--batch-check=%(objectname) %(objecttype) %(objectsize)"]
enableremote judac-storage [2020-05-10 14:01:02.90966] chat: /p/home/.../miniconda3/bin/git-annex-remote-ora []
[2020-05-10 14:01:03.060436] git-annex-remote-ora[1] --> VERSION 1
[2020-05-10 14:01:03.060792] git-annex-remote-ora[1] <-- EXTENSIONS INFO
[2020-05-10 14:01:03.061225] git-annex-remote-ora[1] --> EXTENSIONS
[2020-05-10 14:01:03.061343] git-annex-remote-ora[1] <-- LISTCONFIGS
[2020-05-10 14:01:03.061539] git-annex-remote-ora[1] --> UNSUPPORTED-REQUEST
[2020-05-10 14:01:03.062077] chat: /p/home/.../miniconda3/bin/git-annex-remote-ora []
[2020-05-10 14:01:03.182138] git-annex-remote-ora[1] --> VERSION 1
[2020-05-10 14:01:03.182371] git-annex-remote-ora[1] <-- EXTENSIONS INFO
[2020-05-10 14:01:03.18261] git-annex-remote-ora[1] --> EXTENSIONS
[2020-05-10 14:01:03.182732] git-annex-remote-ora[1] <-- EXPORTSUPPORTED
[2020-05-10 14:01:03.182932] git-annex-remote-ora[1] --> EXPORTSUPPORTED-FAILURE
[2020-05-10 14:01:03.183713] chat: /p/home/.../miniconda3/bin/git-annex-remote-ora []
[2020-05-10 14:01:03.36288] git-annex-remote-ora[1] --> VERSION 1
[2020-05-10 14:01:03.363095] git-annex-remote-ora[1] <-- EXTENSIONS INFO
[2020-05-10 14:01:03.363348] git-annex-remote-ora[1] --> EXTENSIONS
[2020-05-10 14:01:03.363458] git-annex-remote-ora[1] <-- LISTCONFIGS
[2020-05-10 14:01:03.363624] git-annex-remote-ora[1] --> UNSUPPORTED-REQUEST
[2020-05-10 14:01:03.36376] git-annex-remote-ora[1] <-- INITREMOTE
[2020-05-10 14:01:03.363924] git-annex-remote-ora[1] --> GETGITDIR
[2020-05-10 14:01:03.364023] git-annex-remote-ora[1] <-- VALUE .git
[2020-05-10 14:01:03.364176] git-annex-remote-ora[1] --> GETCONFIG name
[2020-05-10 14:01:03.364262] git-annex-remote-ora[1] <-- VALUE 
[2020-05-10 14:01:03.36449] git-annex-remote-ora[1] --> INITREMOTE-FAILURE Cannot determine special remote name, got: ''

git-annex: Cannot determine special remote name, got: ''
failed
[2020-05-10 14:01:03.365193] process done ExitSuccess
[2020-05-10 14:01:03.365697] process done ExitSuccess
git-annex: enableremote: 1 failed
```

Seems git-annex is not working as expected. Also:

```
$ git annex version
git-annex version: 7.20200204-g4db801d
```

So
```
$ conda install git-annex -c conda-forge
$ git annex version
git-annex version: 8.20200309-g07fcace
```

and now things start working, but not as expected:

```
$ git annex enableremote judac-storage
git-annex: ConnectionOpenFailedError: ''['"'"'ssh'"'"', '"'"'-fN'"'"', '"'"'-o'"'"', '"'"'ControlMaster=auto'"'"', '"'"'-o'"'"', '"'"'ControlPersist=15m'"'"', '"'"'-o'"'"', '"'"'ControlPath=/p/home/.../.cache/datalad/sockets/c2f7fe9d'"'"', '"'"'judac.fz-juelich.de'"'"']'' failed with exitcode 255 [Failed to open SSH connection (could not start ControlMaster process)]
failed
git-annex: enableremote: 1 failed
```

(SSH isn't possible from here)

On a full reclone with git-annex 8.x it works

```
$ datalad clone ria+file:///p/largedata/ukbiobank/dataladstores/enki#~bids_raw 
[INFO   ] Scanning for unlocked files (this may take some time)                              
| judac.fz-juelich.de: Permission denied (publickey,gssapi-keyex,gssapi-with-mic). 
[INFO   ] ConnectionOpenFailedError: ''['"'"'ssh'"'"', '"'"'-fN'"'"', '"'"'-o'"'"', '"'"'ControlMaster=auto'"'"', '"'"'-o'"'"', '"'"'ControlPersist=15m'"'"', '"'"'-o'"'"', '"'"'ControlPath=/p/home/.../.cache/datalad/sockets/c2f7fe9d'"'"', '"'"'judac.fz-juelich.de'"'"']'' failed with exitcode 255 [Failed to open SSH connection (could not start ControlMaster process)] 
[INFO   ] Reconfigured judac-storage for ria+file:///p/largedata/.../enki 
[INFO   ] Configure additional publication dependency on "judac-storage" 
configure-sibling(ok): . (sibling)
install(ok): /p/home/.../bids_raw (dataset)
action summary:
  configure-sibling (ok: 1)
  install (ok: 1)
```

Conclusion: we need git-annex 8.x

**Comment by bpoldrack** [comment]
Now that I see this, we've had issues with that annex bug before, I think. Do you remember, @kyleam ?

**Comment by kyleam** [comment]
> Now that I see this, we've had issues with that annex bug before, I think. Do you remember, @kyleam ?

This looks like the issue fixed with 6a9147192 (GETCONFIG name fix, 2020-03-09), which was part of the 8.20200309 release and fixed a regression introduced in 7.20200202.7.  I haven't verified that that's actually what's going on in this example, but it lines up with the reported versions, at least.

**Comment by mih** [comment]
OK, so this looks sufficiently researched and short of adjusting the minimum version of git-annex, there is little that we could or should do. Thx to @bpoldrack and @kyleam for their research.

**Comment by mih** [comment]
FTR: I see this bug using the stock git-annex available from Ubuntu 20.04 (see https://ci.appveyor.com/project/mih/appveyor-datalad/build/job/i7hqqmtve72k4myc), so it will be with us for the next 4-5 years.
